### PR TITLE
 RUST-732 Mark the versioned API options public.

### PR DESCRIPTION
**Pull request by abr-egn**
RUST-732

This approximately reverts the [commit](https://github.com/mongodb/mongo-rust-driver/commit/ffae5c31c6ac368fc0c242c3bcf3b5d8811a7f09) hiding the options; it also updates the documentation for those options to link to the manual (RUST-605).

### CODE REVIEW

File: src/client/options/mod.rs [hunk 1]
```diff
@@ -500,8 +501,11 @@ pub struct ClientOptions {
     /// supported and is considered undefined behaviour. To run any command with a different API
     /// version or without declaring one, create a separate client that declares the
     /// appropriate API version.
-    #[builder(default, setter(skip))]
```
Comment (patrickfreed) [inline src/client/options/mod.rs:503]: Can we also update the builder for `ConnectionPoolOptions` to not skip this field? Alternatively we could just remove the builder API for `ConnectionPoolOptions` as it isn't a public-facing type and can be constructed from a `ClientOptions`, which does have a builder.

File: src/client/options/mod.rs [hunk 2]
```diff
@@ -500,8 +501,11 @@ pub struct ClientOptions {
...
-    pub(crate) server_api: Option<ServerApi>,
+    ///
+    /// For more information, see the [Versioned API](
+    /// https://docs.mongodb.com/v5.0/reference/versioned-api/) manual page.
```
Comment (patrickfreed) [inline src/client/options/mod.rs:506]: Can we add this link to the `ServerApi` struct declaration too?

Comment (abr-egn) [reply src/client/options/mod.rs:506]: Done.

File: src/client/options/mod.rs (hunk 1 above)
Comment (abr-egn) [reply src/client/options/mod.rs:503]: Removed the builder API.

**Review by patrickfreed — APPROVED** [review]
LGTM!

**Review by isabelatkinson — APPROVED** [review]
lgtm, sorry for the late review!

**Pull request merged**